broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExporterContainerTest.java

```diff
@@ -11,53 +11,42 @@
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.when;
 
-import io.camunda.zeebe.broker.exporter.repo.ExporterDescriptor;
+import io.camunda.zeebe.broker.exporter.repo.ExporterLoadException;
 import io.camunda.zeebe.engine.processing.streamprocessor.TypedRecord;
-import io.camunda.zeebe.engine.state.DefaultZeebeDbFactory;
 import io.camunda.zeebe.exporter.api.Exporter;
 import io.camunda.zeebe.exporter.api.context.Context;
 import io.camunda.zeebe.exporter.api.context.Controller;
 import io.camunda.zeebe.protocol.impl.record.RecordMetadata;
 import io.camunda.zeebe.protocol.record.Record;
 import io.camunda.zeebe.protocol.record.RecordType;
 import io.camunda.zeebe.protocol.record.ValueType;
-import io.camunda.zeebe.util.sched.Actor;
-import io.camunda.zeebe.util.sched.ActorControl;
-import io.camunda.zeebe.util.sched.testing.ActorSchedulerRule;
-import java.io.IOException;
+import java.nio.file.Path;
 import java.util.Map;
-import org.junit.Before;
-import org.junit.Rule;
-import org.junit.Test;
-import org.junit.rules.TemporaryFolder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import org.junit.jupiter.api.parallel.Execution;
+import org.junit.jupiter.api.parallel.ExecutionMode;
 
-public class ExporterContainerTest {
+@Execution(ExecutionMode.CONCURRENT)
+final class ExporterContainerTest {
 
-  @Rule public final ActorSchedulerRule actorSchedulerRule = new ActorSchedulerRule();
-  @Rule public final TemporaryFolder tempFolder = new TemporaryFolder();
-  private TestActor testActor;
-  private ExportersState exportersState;
+  private ExporterContainerRuntime runtime;
   private FakeExporter exporter;
   private ExporterContainer exporterContainer;
 
-  @Before
-  public void setup() throws IOException {
-    testActor = new TestActor();
-    actorSchedulerRule.submitActor(testActor).join();
-    final var dbFactory = DefaultZeebeDbFactory.defaultFactory();
-    final var db = dbFactory.createDb(tempFolder.newFolder());
-    exportersState = new ExportersState(db, db.createContext());
-    final ExporterMetrics exporterMetrics = new ExporterMetrics(1);
-
-    final var exporterDescriptor =
-        new ExporterDescriptor("fakeExporter", FakeExporter.class, Map.of("key", "value"));
-    exporterContainer = new ExporterContainer(exporterDescriptor);
+  @BeforeEach
+  void beforeEach(final @TempDir Path storagePath) throws ExporterLoadException {
+    runtime = new ExporterContainerRuntime(storagePath);
+
+    final var descriptor =
+        runtime.getRepository().load("fakeExporter", FakeExporter.class, Map.of("key", "value"));
```

Comment:
Actually, this method really should be in the repository test class, which is why it seems a little strange here. There we just work directly with the repository - so it might make more sense to move it there. I'll think about it a bit